### PR TITLE
Fixes 3463 replace assert with assert that

### LINKED ISSUE #3463: Remove usages of assert
```
/**
 * Throws an [AssertionError] if the [value] is false
 * and runtime assertions have been enabled on the JVM using the *-ea* JVM option.
 */
@kotlin.internal.InlineOnly
public inline fun assert(value: Boolean) {
    assert(value) { "Assertion failed" }
}
```
*and runtime assertions have been enabled on the JVM using the *-ea* JVM option.* 
They have not been enabled and won't be. Replace with appropriate `AssertThat`s.

### PR DESCRIPTION
**Pull request by u7281898**
**Description (required)**

Fixes #3463

What changes did you make and why?

I replaced all instances of assert() with assertThat() where appropriate. 
assertThat() error messages are easier to understand and the assertThat method has a wider range of abilities to determine if the output of a function is appropriate. For example assertThat will not complete compiling instead of failing if the expected input is an integer instead of a string.

**Tests performed (required)**

Tested AccountUtilUnitTest with Pixel 2 API 30

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java [hunk 1]
```diff
@@ -991,6 +995,7 @@ public void onActivityResult(final int requestCode, final int resultCode,
 
         if (requestCode == REQUEST_CODE && resultCode == RESULT_OK) {
 
+//            assertThat(data, not(nullValue()));
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java:998]: Is this really needed?

File: data-client/build.gradle [hunk 2]
```diff
@@ -32,11 +32,11 @@ version = "${VERSION_NAME}"
 group = "${GROUP_ID}"
 
 android {
-    compileSdkVersion 28
+    compileSdkVersion 30
 
     defaultConfig {
         minSdkVersion 19
-        targetSdkVersion 28
+        targetSdkVersion 30
```
Comment (nicolas-raoul) [inline data-client/build.gradle:39]: Is this really needed?

File: app/src/test/kotlin/fr/free/nrw/commons/auth/AccountUtilUnitTest.kt [hunk 3]
```diff
@@ -3,13 +3,16 @@ package fr.free.nrw.commons.auth
 import fr.free.nrw.commons.FakeContextWrapper
 import fr.free.nrw.commons.FakeContextWrapperWithException
 import fr.free.nrw.commons.TestCommonsApplication
-import org.junit.Assert
+import org.junit.Assert.*
```
Comment (nicolas-raoul) [inline app/src/test/kotlin/fr/free/nrw/commons/auth/AccountUtilUnitTest.kt:6]: Would you mind importing only the needed classes, explicitly?

File: app/src/test/kotlin/fr/free/nrw/commons/auth/AccountUtilUnitTest.kt [hunk 4]
```diff
@@ -3,13 +3,16 @@ package fr.free.nrw.commons.auth
...
 import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 import org.robolectric.RobolectricTestRunner
 import org.robolectric.RuntimeEnvironment
 import org.robolectric.annotation.Config
+import org.hamcrest.CoreMatchers.*;
```
Comment (nicolas-raoul) [inline app/src/test/kotlin/fr/free/nrw/commons/auth/AccountUtilUnitTest.kt:13]: Would you mind importing only the needed classes, explicitly?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Looks good generally, just a few minor things to improve.

**Comment by domdomegg** [comment]
Closing this as now stale

**Pull request closed without merging**